pytorch_lightning/profiler/pytorch.py

```diff
@@ -43,55 +101,35 @@ class PyTorchProfiler(BaseProfiler):
         "self_cuda_memory_usage",
         "count",
     )
+    START_RECORD_FUNCTIONS = ('on_train_start', 'on_validation_step', 'on_test_start', 'on_predict_start')
 
     def __init__(
         self,
         output_filename: Optional[str] = None,
-        enabled: bool = True,
-        use_cuda: bool = False,
-        record_shapes: bool = False,
-        profile_memory: bool = False,
         group_by_input_shapes: bool = False,
-        with_stack: bool = False,
-        use_kineto: bool = False,
-        use_cpu: bool = True,
         emit_nvtx: bool = False,
-        export_to_chrome: bool = False,
-        path_to_export_trace: str = None,
+        export_to_chrome: bool = True,
+        path_to_export_trace: Optional[str] = None,
         row_limit: int = 20,
         sort_by_key: Optional[str] = None,
-        profiled_functions: Optional[List] = None,
+        record_functions: List[str] = [],
         local_rank: Optional[int] = None,
-    ):
+        profiled_functions: List[str] = [],
```

Comment:
let's not have mutable default values in the constructor. we can make them Optional and set the default inside the `__init__`

Reply:
Done !